app-modules/ai/resources/views/components/assistant.blade.php

```diff
@@ -89,6 +89,19 @@ class="flex flex-col gap-y-1 rounded-xl border border-gray-950/5 bg-white p-2 sh
                         @foreach ($this->threadsWithoutAFolder as $threadItem)
                             <li
                                 id="chat-{{ $threadItem->id }}"
+                                title="Last Engaged: {{ $threadItem?->last_engaged_at?->format('F d, Y') }}"
+                                x-data="{
+                                    updateTitle(time, threadItemId) {
+                                            const formattedDate = new Date(time).toLocaleDateString('en-US', { year: 'numeric', month: 'long', day: 'numeric' });
+                                            document.getElementById('chat-' + threadItemId).title = 'Last Engaged: ' + formattedDate;
+                                        },
+                                        listenForEvent(threadId) {
+                                            window.addEventListener(`message-sent-${threadId}`, (event) => {
+                                                this.updateTitle(event.detail.time, event.detail.threadItemId);
+                                            });
+                                        }
+                                }"
+                                x-init="listenForEvent('{{ $threadItem->id }}')"
```

Comment:
Now that I am looking at this all together I am realizing that instead of placing this `x-data` and `x-init` on each individual thread, you could place this way up the container for all threads. (in or out of a folder)

You can then just listen for one event "message-sent" and then do your `update-title` method.

I was wrong, we don't need to make the event specific per Thread, because you pass the thread ID to you updateTitle and then update all threads with the ID of that thread with the JS.

So I would move this to the container for all threads. And make it so you are just listening to the single event and then updating the proper thread title based on the ID.

Reply:
I think there's quite a lot of Alpine being underutilised here. The event listener could use `x-on` instead of being manually registered in `x-init`. Also, the title could be bound to with `x-bind:title` instead of manually being set through JavaScript. And then the last updated date can be stored in an Alpine property, which makes everything that listens to it reactive when it is updated.

I don't think the event should contain the current time, that could be simplified by getting the current time inside the listener.

I think that potentially, `x-tooltip` should be used instead of `x-bind:title` to make it more visible.